src/python/pants/jvm/jdk_rules.py

```diff
@@ -72,7 +83,7 @@ def immutable_input_digests(self) -> dict[str, Digest]:
 
 
 @dataclass
-class JdkSetup:
+class InternalJdk:
     jdk: JdkEnvironment
```

Comment:
Rather than a wrapper, it seems like this could maybe be a subclass...? That would allow for directly using it with `JdkProcess`. Seems like a relative safe use of OO maybe.

Reply:
Nice. I was getting somewhat stuck in the weeds of constructing the `InternalJdk` object, but yeah, it doesn't need to be done cleverly, does it?